pkg/autodiscovery/autoconfig.go

```diff
@@ -469,15 +499,78 @@ func (ac *AutoConfig) pollConfigs() {
 func (ac *AutoConfig) processRemovedConfigs(configs []integration.Config) {
 	ac.scheduler.Unschedule(configs)
 	for _, c := range configs {
-		delete(ac.configResolver.configToService, c.Digest())
 		ac.store.removeLoadedConfig(c)
 		// if the config is a template, remove it from the cache
 		if c.IsTemplate() {
-			ac.templateCache.Del(c)
+			ac.store.templateCache.Del(c)
 		}
 	}
 }
 
+// resolveTemplate attempts to resolve a configuration template using the AD
+// identifiers in the `integration.Config` struct to match a Service.
+//
+// The function might return more than one configuration for a single template,
+// for example when the `ad_identifiers` section of a config.yaml file contains
+// multiple entries, or when more than one Service has the same identifier,
+// e.g. 'redis'.
+//
+// The function might return an empty list in the case the configuration has a
+// list of Autodiscovery identifiers for services that are unknown to the
+// resolver at this moment.
+func (ac *AutoConfig) resolveTemplate(tpl integration.Config) []integration.Config {
+	// use a map to dedupe configurations
+	// FIXME: the config digest as the key is currently not reliable
```

Comment:
What are you suggesting to do though ?

Reply:
This was an existing FIXME, we have some backlog work to fix this.